src/ThreadPool.zig

```diff
@@ -30,49 +30,37 @@ pub fn init(self: *ThreadPool, allocator: *std.mem.Allocator) !void {
 
     errdefer self.deinit();
 
-    var num_threads = std.Thread.cpuCount() catch 1;
-    if (num_threads > 0)
-        self.threads = try allocator.alloc(*std.Thread, num_threads);
+    var num_threads = std.math.max(1, std.Thread.cpuCount() catch 1);
+    self.threads = try allocator.alloc(*std.Thread, num_threads);
 
     while (num_threads > 0) : (num_threads -= 1) {
         const thread = try std.Thread.spawn(self, runWorker);
-        self.threads[self.running] = thread;
-        self.running += 1;
+        self.threads[self.spawned] = thread;
+        self.spawned += 1;
     }
```

Comment:
In the loop both `num_threads` and `spawned` are being used as counters, but only one is needed.  `num_threads` can be const and we could just use `spawned` as the counter:
```zig
    const num_threads = std.math.max(1, std.Thread.cpuCount() catch 1);
    self.threads = try allocator.alloc(*std.Thread, num_threads);

    while (self.spawned < num_threads) : (self.spawned += 1) {
        const thread = try std.Thread.spawn(self, runWorker);
        self.threads[self.spawned] = thread;
    }
```

Reply:
Changed it, thanks.